### PR TITLE
Fix to cluster-state-service getting stale after networking issue.

### PR DESCRIPTION
**Pull request by kylbarnes**
#### Summary
Fix to cluster-state-service getting stale after networking issue.

#### Description
This pull request addresses the cluster-state-service getting stale when encountering a networking issue. We previously had no timeout values set on our SQS consumer, which would cause it to block indefinitely on any networking issues. This change sets the http timeout to 30s, which allows the sqs consumer to recover and continue polling.

#### Implementation details
* Changed http transport to http.DefaultTransport.
* Set http client timeout to 30 seconds.

#### Testing
- [x] cluster-state-service binary built locally and unit-tests pass (`cd cluster-state-service; make; cd ../`)
- [x] cluster-state-service build in Docker succeeds (`cd cluster-state-service; make release; cd ../`)
- [x] daemon-scheduler binary built locally and unit-tests pass (`cd daemon-scheduler; make; cd ../`)
- [x] daemon-scheduler build in Docker succeeds (`cd daemon-scheduler; make release; cd ../`)

New tests cover the changes: no

#### Description for the changelog
Fix to cluster-state-service getting stale after networking issue

#### Licensing
This contribution is under the terms of the Apache 2.0 License: Yes

#### Before merging
- [x] cluster-state-service end-to-end tests pass. Required setup details are listed [here](https://github.com/blox/blox/blob/dev/cluster-state-service/internal/Readme.md).
- [x] daemon-scheduler integration tests pass. Required setup details are listed [here](https://github.com/blox/blox/blob/dev/daemon-scheduler/internal/features/README.md).
- [x] daemon-scheduler end-to-end tests pass. Required setup details are listed [here](https://github.com/blox/blox/blob/dev/daemon-scheduler/internal/features/README.md).

### CODE REVIEW

**Comment by kylbarnes** [comment]
This pull request addresses the cluster-state-service getting stale when encountering a networking issue. We previously had no timeout values set on our SQS consumer, which would cause it to block indefinitely on any networking issues. This change sets the http timeout to 30s, which allows the sqs consumer to recover and continue polling.

File: cluster-state-service/handler/httpclient/httpclient.go [hunk 1]
```diff
@@ -47,10 +51,11 @@ func init() {
 
 // New returns an Blox httpClient that will insert custom HTTP UA header.
 func New() *http.Client {
-	transport := &http.Transport{}
+	transport := http.DefaultTransport
 
 	client := &http.Client{
 		Transport: &bloxRoundTripper{transport},
+		Timeout: httpClientTimeout,
```
Comment (poojamaiya) [inline cluster-state-service/handler/httpclient/httpclient.go]: Out of curiosity, what is the different between this and the timeout in http.DefaultTransport?

**Comment by poojamaiya** [comment]
@kylbarnes, I'd add your comment `This pull request addresses the cluster-state-service getting stale when encountering a networking issue. We previously had no timeout values set on our SQS consumer, which would cause it to block indefinitely on any networking issues. This change sets the http timeout to 30s, which allows the sqs consumer to recover and continue polling.` to the description of the PR.

File: cluster-state-service/handler/httpclient/httpclient.go (hunk 1 above)
Comment (kylbarnes) [inline cluster-state-service/handler/httpclient/httpclient.go]: This is a hard timeout. The timeouts in `http.DefaultTransport` are more granular. It sets the keep-alive idle timeout to 90s and the connection timeout to 30s. This means that together the individual timeouts in `http.DefaultTransport` can actually exceed 30 seconds. By setting it on the `http.Client`, I am ensuring that the request should timeout if it takes more than 30s from the start, regardless of how those 30s are consumed.

**Comment by kylbarnes** [comment]
@poojamaiya Added to the PR description and removed my comment.

**Pull request merged**